packages/shared/invokeGuardedCallback.js

```diff
@@ -106,6 +111,16 @@ if (__DEV__) {
         // nested call would trigger the fake event handlers of any call higher
         // in the stack.
         fakeNode.removeEventListener(evtType, callCallback, false);
+
+        // We check for window.hasOwnProperty('event') to prevent the window.event assignment
+        // in IE as it crashes in strict mode.
+        if (
+          typeof window.event !== 'undefined' &&
+          window.hasOwnProperty('event')
```

Comment:
What's the story behind this?
Why does `window.hasOwnProperty('event')` tell us whether it's *writable*?

Reply:
@gaearon Thanks for looking into this.

It is useful for two reasons, only one of which I added to the comment:

1. Running `window.event = event` on IE <= 10 throws an error `"SCRIPT3: Member Not Found"`. As far as I can tell, this has nothing to do with anything other than "running in IE <= 10". So I needed at least a feature check for these browsers.
2. Firefox does not support `window.event` at all. So I wanted a check for browsers that usually have an event property on window.

By happy coincidence this check does both. I've updated the comment, and rebased over your latest changes.

Please let me know if I can improve this more, as this is currently the main thing holding us back on 16.2.